src/bptl/work_units/zgw/tasks/resultaat.py

```diff
@@ -36,22 +40,61 @@ class CreateResultaatTask(ZGWWorkUnit):
     * ``resultaatUrl``: the full URL of the created RESULTAAT
     """
 
+    def _get_resultaattype(self, zaak_url: str, variables: dict) -> str:
+        try:
+            resultaattype = check_variable(variables, "resultaattype")
+        except MissingVariable:
+            try:
+                omschrijving = check_variable(variables, "omschrijving")
+                zrc_client = self.get_client(APITypes.zrc)
+                zaak = zrc_client.retrieve("zaak", url=zaak_url)
+                ztc_client = self.get_client(APITypes.ztc)
+                resultaattypen = ztc_client.list(
+                    "resultaattype",
+                    request_kwargs={"params": {"zaaktype": zaak["zaaktype"]}},
+                )
+                if resultaattypen["count"] == 0:
+                    raise ValueError(
+                        "No resultaattypen were found for zaaktype %s."
+                        % zaak["zaaktype"]
+                    )
+                resultaattype = list(
+                    filter(
+                        lambda rt: rt["omschrijving"].lower() == omschrijving.lower(),
+                        resultaattypen["results"],
+                    )
+                )
+                if len(resultaattype) != 1:
+                    raise ValueError(
+                        "No%s resultaattype was found with zaaktype %s and omschrijving %s."
+                        % (
+                            "" if len(resultaattype) == 0 else " unique",
+                            zaak["zaaktype"],
+                            omschrijving,
+                        )
+                    )
+                resultaattype = resultaattype[0]["url"]
```

Comment:
This belongs in the `else` part of

```
try:
     ...
except ...:
    ...
else:
    ...
```

The entire code block after the `check_variable` should not raise a `MissingVariable`, so it should be outside of this specific try-except block.

Reply:
Thanks!